meta-balena-common/recipes-core/systemd/disk-watchdog/disk-watchdog.c

```diff
@@ -0,0 +1,296 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <getopt.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <systemd/sd-daemon.h>
+
+#define BUFFER_SIZE 512
```

Comment:
Why 512? Would it be better to actually read the sector size as this can vary?

Reply:
I'll copy your comments to the other PR, and work on them there! Thanks for the review